chunk 1: Tweak section headers (local change).
chunk 2: For .TH "", set the title to "UNTITLED" (to be pushed upstream)
chunk 3: Set the default operating system name (local change).
xhunk 4: Fix regression of .OP without arguments (to be pushed upstream).
chunk 5-7: Fix .ME and .UE tail argument regression (to be pushed upstream).

Index: tmac/an.tmac
--- tmac/an.tmac.orig
+++ tmac/an.tmac
@@ -286,7 +286,7 @@
 .  ds an*section1 General Commands Manual\"
 .  ds an*section2 System Calls Manual\"
 .  ds an*section3 Library Functions Manual\"
-.  ds an*section4 Kernel Interfaces Manual\"
+.  ds an*section4 Device Drivers Manual\"
 .  ds an*section5 File Formats Manual\"
 .  ds an*section6 Games Manual\"
 .  ds an*section7 Miscellaneous Information Manual\"
@@ -384,7 +384,8 @@
 .  nr an*is-in-example 0
 .  nr an*is-in-link-text-diversion 0
 .
-.  ds an*ident "\\$1\"
+.  ie '\\$1'' .ds an*ident "UNTITLED\"
+.  el         .ds an*ident "\\$1\"
 .  if \\n[CT] .stringup an*ident
 .  ds an*section "\\$2\"
 .  ie (\\n[.$] > 4) .ds an*extra3 "\\$5\"
@@ -398,7 +399,7 @@
 .
 .  ds an*extra1 "\\$3\"
 .  ie (\\n[.$] > 3)    .ds an*extra2 "\\$4\"
-.  el                  .ds an*extra2 \" empty; but .AT/.UC can override
+.  el                  .ds an*extra2 "OpenBSD\"
 .
 .  if '\\*[an*extra1]'' \{\
 .    ds an*msg .\\$0 missing third argument; consider document\"
@@ -1166,7 +1167,7 @@ contains unsupported escape sequence
 .  ie (\\n[.$] > 1) \
 .    RI [\\f[B]\\$1\f[] \~\\$2 ]
 .  el \
-.    RB [ \\$1 ]
+.    RB [ "\\$1" ]
 ..
 .
 .\" Begin an example (typically of source code or shell input).
@@ -1270,7 +1271,7 @@ contains unsupported escape sequence
 .\" "mailto:", but this need not be visible when rendering an email
 .\" address on a device incapable of hyperlinking.
 .de an*end-hyperlink
-.  ds an*trailing-text \\$1\"
+.  ds an*trailing-text \\$*\"
 .  ie (\\n[an*is-in-link-text-diversion] & \\n[an*do-hyperlink]) \{\
 .    br
 .    di
@@ -1357,7 +1358,7 @@ contains unsupported escape sequence
 .    an*warn .\\$0 requires a preceding .MT
 .    return
 .  \}
-.  an*end-hyperlink \\$1
+.  an*end-hyperlink \\$*
 .  rm an*prefix
 ..
 .
@@ -1383,7 +1384,7 @@ contains unsupported escape sequence
 .    an*warn .\\$0 requires a preceding .UR
 .    return
 .  \}
-.  an*end-hyperlink \\$1
+.  an*end-hyperlink \\$*
 .  rm an*prefix
 ..
 .
